core/species.cpp

```diff
@@ -5486,131 +6101,165 @@ void Species::WriteTreeSequence(std::string &p_recording_tree_path, bool p_simpl
 	// it *does* change the order of the rows; see https://github.com/MesserLab/SLiM/issues/209
 	if (p_simplify)
 	{
-		SimplifyTreeSequence();
+		SimplifyAllTreeSequences();
 	}
 	
-	// Copy the table collection so that modifications we do for writing don't affect the original tables
-	tsk_table_collection_t output_tables;
-	ret = tsk_table_collection_copy(&tables_, &output_tables, 0);
-	if (ret < 0) handle_error("tsk_table_collection_copy", ret);
-	
-	// Sort and deduplicate; we don't need to do this if we simplified above, since simplification does these steps
-	if (!p_simplify)
+	for (Chromosome *chromosome : chromosomes_)
 	{
-		int flags = TSK_NO_CHECK_INTEGRITY;
+		slim_chromosome_index_t chromosome_index = chromosome->Index();
+		TreeSeqInfo &chromosome_tsinfo = treeseq_[chromosome_index];
+		tsk_table_collection_t &chromosome_tables = chromosome_tsinfo.tables_;
+		
+		// Copy in the shared tables (node, individual, population) at this point, so the shared tables then get
+		// copied below; we will be modifying the tables, and don't want our modification to go into the original
+		// shared tables, which we are not allowed to change.
+		if (chromosome_index > 0)
+			CopySharedTablesIn(chromosome_tables);
+		
+		// Copy the table collection so that modifications we do for writing don't affect the original tables.
+		// Note that there's a lot of work below to clean up the individuals table and node table for saving.
+		// Those tables are shared.  We don't want to do this cleanup in the original tables, since that would
+		// modify our recording state I guess; but I think this cleanup will be the same for every chromosome,
+		// so technically we could do this work just once, I think (?), and share the processed tables across
```

Comment:
yes, this is definitely true, because we want the shared tables to be identical on output!

Reply:
Hmm.  I was thinking that this was just a performance issue – that we would save some (probably very small) amount of time by doing the work just once.  I figured the work is deterministic and will produce the same result every time, from the same data, and so "identical on output" would be true in any case.  But are you saying that you think that is not the case, and that it is therefore important to do the work just once, and not just for performance reasons?  If so, why?

Reply:
Ah, I see you write more on this below...

Reply:
Nope - I agree that it should be deterministic, and if it is not then there is A Problem; but it seems like the code is cleaner if the work is done once? Maybe not?

Reply:
Maybe.  I didn't see a clear/simple path to doing the work once.  I'll think about it more.

Reply:
resolving, will not fix (I agree with myself that it would be hard to fix this cleanly, without increasing the high-water memory usage mark, and it is not important I don't think)